lib/liquid/template.rb

```diff
@@ -183,20 +183,15 @@ def render(*args)
       when Hash
         options = args.pop
 
-        if options[:registers].is_a?(Hash)
-          registers.merge!(options[:registers])
-        end
+        registers.merge!(options[:registers]) if options[:registers].is_a?(Hash)
 
-        if options[:filters]
-          context.add_filters(options[:filters])
-        end
+        context.add_filters(options[:filters]) if options[:filters]
 
-        if options[:exception_handler]
-          context.exception_handler = options[:exception_handler]
-        end
-      when Module
-        context.add_filters(args.pop)
-      when Array
+        context.add_global_filters(options[:global_filters]) if options[:global_filters]
```

Comment:
I agree on the name conflict @dylanahsmith. Actually I don't really understand why the current Liquid::Strainer.global_filter is called what it called. I believe these are custom filters that have not been registered:
Liquid::Template.register_filter(TextFilter)
but rather just passed in at render time:
@template.render({}, :filters => [TextFilter])

Just a suggestion, but maybe they should be called custom_filters or registered_filters?

The reason I went with the 'global' word was to capture the idea that this filter was being applied to all variables in the template.